keg/generator.py

```
@@ -19,69 +19,98 @@
+            )
+        self.image_definition = image_definition
+        self.description_schemas = os.path.join(
+            image_definition.recipes_root, 'schemas'
+        )
+        self.dest_dir = dest_dir
+        self.env = Environment(
+            loader=FileSystemLoader(self.description_schemas)
+        )
+        self.env.globals['keg_funcs'] = template_functions
 
-    img = image_definition.ImageDefinition(image_source, recipes_root, data_roots)
-    img.populate()
+        self.image_definition.populate()
 
-    schemas_dir = os.path.join(recipes_root, 'schemas')
-    env = Environment(loader=FileSystemLoader(schemas_dir))
-    env.globals['keg_funcs'] = template_functions
+    def create_kiwi_description(
+        self, markup: str = 'xml', override: bool = False
+    ):
+        """
+        Creates KIWI config.xml from a KegImageDefinition.
 
-    kiwi_templ = env.get_template('{}.kiwi.templ'.format(img['schema']))
-    kiwi_doc = kiwi_templ.render(data=img.get_data())
+        :param bool override:
+            Override destination contents, default is: False
+        """
+        outfile = os.path.join(self.dest_dir, 'config.kiwi')
+        self._validate_outfile(outfile, override)
 
-    outfile = os.path.join(dest_dir, 'config.kiwi')
-    if os.path.exists(outfile) and not force:
-        raise KegError('{} exists, use force to overwrite.'.format(outfile))
-    with open(outfile, 'w') as fd:
-        fd.write(kiwi_doc)
+        kiwi_template = self.env.get_template(
+            '{}.kiwi.templ'.format(self.image_definition.data['schema'])
+        )
+        kiwi_document = kiwi_template.render(
+            data=self.image_definition.data
+        )
+        with open(outfile, 'w') as kiwi_config:
+            kiwi_config.write(kiwi_document)
+        kiwi = KiwiDescription(outfile)
+        if markup == 'xml':
+            kiwi.create_XML_description(outfile)
+        if markup == 'yaml':
+            kiwi.create_YAML_description(outfile)
```

I think we  should add an `else:` option to log | print a message informing that that markup is not supported.

---

yes I think it should raise an error in this case

---

done